src/shell/js/composeui-fdc3/src/ComposeUIDesktopAgent.ts

```diff
@@ -0,0 +1,308 @@
+/* 
+ *  Morgan Stanley makes this available to you under the Apache License,
+ *  Version 2.0 (the "License"). You may obtain a copy of the License at
+ *       http://www.apache.org/licenses/LICENSE-2.0.
+ *  See the NOTICE file distributed with this work for additional information
+ *  regarding copyright ownership. Unless required by applicable law or agreed
+ *  to in writing, software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ *  or implied. See the License for the specific language governing permissions
+ *  and limitations under the License.
+ *  
+ */
+
+import {
+    AppIdentifier,
+    AppIntent,
+    AppMetadata,
+    Channel,
+    ChannelError,
+    Context,
+    ContextHandler,
+    DesktopAgent,
+    ImplementationMetadata,
+    IntentHandler,
+    IntentResolution,
+    Listener,
+    PrivateChannel
+} from '@finos/fdc3'
+import { MessageRouter, TopicMessage } from '@morgan-stanley/composeui-messaging-client';
+import { ComposeUIChannel } from './infrastructure/ComposeUIChannel';
+import { ChannelType } from './infrastructure/ChannelType';
+import { ComposeUIListener } from './infrastructure/ComposeUIListener';
+import { Fdc3FindChannelRequest } from './infrastructure/messages/Fdc3FindChannelRequest';
+import { Fdc3FindChannelResponse } from './infrastructure/messages/Fdc3FindChannelResponse';
+import { ComposeUITopic } from './infrastructure/ComposeUITopic';
+
+export class ComposeUIDesktopAgent implements DesktopAgent {
+    private appChannels: ComposeUIChannel[] = [];
+    private userChannels: ComposeUIChannel[] = [];
+    private privateChannels: ComposeUIChannel[] = [];
+    private currentChannel?: ComposeUIChannel;
```

Comment:
Simplify as `currentChannel?: ComposeUIChannel` and check any warnings. If something is just optional, you don't need `null` in the type spec.